benchmarks/experiment_runner.py

```diff
@@ -414,9 +414,14 @@ def _prepare_inputs(self, example_inputs, should_randomize_input):
       inputs_list.append(inputs)
     return inputs_list
 
-  def _mark_step(self, benchmark_experiment):
+  def _mark_step(self, benchmark_experiment, tensor_to_check=None):
     if benchmark_experiment.xla:
-      xm.mark_step()
+      if benchmark_experiment.use_torch_xla2:
+        assert tensor_to_check is not None, "torch_xla2 requires input tensor to block_until_ready"
```

Comment:
nit: do you need the assert? Or to block_until_ready on each tensor passed in

Reply:
Oh, we may still need that because `tensor_to_check ` can be None and will error out in the for statement.